internal/util/fscrypt/fscrypt.go

```diff
@@ -94,14 +95,35 @@ func getPassphrase(ctx context.Context, encryption util.VolumeEncryption, volID
 	return passphrase, nil
 }
 
+// resizePassphrase makes sure that the given passphrase will be of [size]
+// bytes. In case the passphrase is shorter, it will be repeated as many times
+// as needed. When a passphrase is (or becomes) longer than the requested
+// [size], the passphrase in truncated.
+func resizePassphrase(passphrase string, size int) (string, error) {
```

Comment:
i assume this function is added to make sure we always get the size of 32 which is required what will happen if the passphrase size is 8 or something even if its added it will not become 32?

Reply:
See the added unit test `TestResizePassphrase`, I think it works as described, no matter what size of password is used.

Reply:
Can we add a case where the passphrase length is 3 and the size is 7 and validate returned passphrase length is matching the size?

Reply:
Sure, added them now.